src/azul/uuids.py

```diff
@@ -0,0 +1,73 @@
+import uuid
+
+
+class InvalidUUIDError(Exception):
+
+    def __init__(self, entity_id: str):
+        super().__init__(f'{entity_id} is not a valid UUID.')
+
+
+class InvalidUUIDVersionError(InvalidUUIDError):
+
+    def __init__(self, uuid: uuid.UUID):
+        Exception.__init__(self, f'{str(uuid)} is a version {uuid.version} UUID (not standardized).')
+
+
+class InvalidUUIDPrefixError(Exception):
+
+    def __init__(self, prefix: str):
+        super().__init__(f'{prefix} is not a valid standard UUID prefix.')
+
+
+def validate_uuid(uuid_str: str) -> uuid.UUID:
+    """
+    >>> validate_uuid('foo')
+    Traceback (most recent call last):
+    ...
+    azul.uuids.InvalidUUIDError: foo is not a valid UUID.
+
+    >>> validate_uuid('8f53d355-b2fa-4bab-a2f2-6852d852d2ec')
+    UUID('8f53d355-b2fa-4bab-a2f2-6852d852d2ec')
+
+    >>> validate_uuid('aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa')
+    Traceback (most recent call last):
+    ...
+    azul.uuids.InvalidUUIDVersionError: aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa is a version 10 UUID (not standardized).
+    """
+    try:
+        formatted_uuid = uuid.UUID(uuid_str)
+    except ValueError:
+        raise InvalidUUIDError(uuid_str)
+    else:
+        if str(formatted_uuid) != uuid_str:
+            raise InvalidUUIDError(uuid_str)
+        if formatted_uuid.version is None or not (1 <= formatted_uuid.version <= 4):
```

Comment:
```suggestion
        if formatted_uuid.version not in (1, 3, 4, 5):
```

First paragraph in https://docs.python.org/3/library/uuid.html